### PR TITLE
Removes OSX from normal PR matrix

### LINKED ISSUE #1333: Move both Travis + Github Actions OSX jobs to nightlies
## Problem Definition

So @kelsos as we discussed in our meeting today, it's a waste of CI cycles and dev time to be running the OSX jobs that take 3x the time of the rest of the CI in every PR. Instead we should just run them nightly.

## Task

Do it.

### PR DESCRIPTION
**Pull request by kelsos**
Closes #1333 
I already created a cron on travis

Also it seems that having conditional matrix execution is not easy, so a new workflow was created for the `nightlies`

### CODE REVIEW

File: .github/workflows/rotki_nightly.yml [hunk 1]
```diff
@@ -0,0 +1,189 @@
+name: Rotki Nightly
```
Comment (LefterisJP) [inline .github/workflows/rotki_nightly.yml:1]: So we will need to keep this in sync with any changes that happen to the rotki_ci.yml workflow?

File: .travis.yml [hunk 2]
```diff
@@ -156,18 +145,12 @@ jobs:
             - $GENERATED_APPIMAGE
             - $APPIMAGE_CHECKSUM
             - $TAR_CHECKSUM
-          skip_cleanup: true
```
Comment (LefterisJP) [inline .travis.yml:159]: why remove this?

File: .travis.yml [hunk 3]
```diff
@@ -156,18 +145,12 @@ jobs:
...
           draft: true
           on:
             tags: true
             condition: $TRAVIS_OS_NAME = linux
             repo: rotki/rotki
-            all_branch: true
-      on:
```
Comment (LefterisJP) [inline .travis.yml:166]: why remove this on condition?

File: .github/workflows/rotki_nightly.yml (hunk 1 above)
Comment (kelsos) [reply .github/workflows/rotki_nightly.yml:1]: Unfortunately yes :'(

File: .travis.yml (hunk 2 above)
Comment (kelsos) [reply .travis.yml:159]: This is now the default behavior, and this flag has been deprecated. There is a new flag if you want to clean up.

**Pull request merged**